iorgen/types.py

```diff
@@ -409,6 +409,8 @@ def from_dict(cls: T[INPUT], dic: Dict[str, Any]) -> Optional[INPUT]:
             if "function_name" not in dic and "name" in dic:
                 print('WARNING: "name" is deprecated, use "function_name"')
                 dic["function_name"] = dic["name"]
+            if not re.fullmatch('[a-z][a-z0-9 ]*', dic['function_name']):
+                raise ValueError('Field `function_name` should match [a-z][a-z0-9 ]*')
```

Comment:
For the rest of the code, I avoided exceptions and returned `None` instead. And then analyzed what went wrong in `checkinput.py`. But I do not like how I did things, so we can keep this `raise` here.
Probably in the future I’ll replace all the `return None` by a `raise`.

Reply:
I saw it but it would lead to unnecessary code duplication.